src/Razor/Microsoft.NET.Sdk.Razor/src/build/netstandard2.0/Microsoft.NET.Sdk.Razor.Components.Wasm.targets

```diff
@@ -0,0 +1,482 @@
+<!--
```

Comment:
Since there's no diff associated with this file, it's pretty hard to CR and reason about what's changing.

Do you think you'd be able to summarize the high-level points about what's being done differently here now?

Reply:
The "Build" part of the pipeline is pretty close to what we had in Blazor.MonoRuntime.targets. We gather inputs, generate a blazor.boot.json, optionally a server worker manifest, add things to StaticWebAssets, and copy files to `$(OutputPath)wwwroot`. We no longer run the linker or compression during, so those parts of the pipeline are removed.

When publishing standalone, the linker operates on `ResolvedFileToPublish`. These files are inputs to compression, published PWA manifest, and the published blazor.boot.json. It's pretty much like build, but with different inputs. To make our lives easier, rather than the targets calculating values (such as `BootManifestResourceType`), the tasks do it. This saves us from duplicating a lot of MSBuild code between the build and publish targets.

 Publishing P2P is a bit more interesting. A `dotnet publish` on a hosted server project, only results in a `dotnet build` on the referenced client project. This is how MSBuild works. But we want this to produce a "published" WASM app. There is an escape hatch though. MSBuild also calls `GetCopyToPublishDirectoryItems` on referenced projects to get files to copy to the server's outputs. We use this to do a publish on the client project and copy them over to the server.